src/detect-dns-answer-name.c

```diff
@@ -56,6 +56,12 @@ static InspectionBuffer *GetBuffer(DetectEngineThreadCtx *det_ctx,
         return buffer;
     }
 
+    if (f->alproto == ALPROTO_DOH2) {
```

Comment:
can we hide this behind this a single (inline?) call in all places?

Reply:
Will try to make it nicer...